fili-core/src/main/java/com/yahoo/bard/webservice/druid/serializers/Util.java

```diff
@@ -44,4 +45,25 @@
         // If we cannot find the physical name, then return empty optional.
         return Optional.empty();
     }
+
+    /**
+     * JSON tree walk to determine if there is a nested query below the current json node or not.
+     *
+     * @param gen  the Json Generator to retrieve the tree to walk on
+     *
+     * @return  a Boolean where true indicates there are more nested query below this node, false otherwise
+     */
+    public static Boolean moreNestedQuery(JsonGenerator gen) {
+        JsonStreamContext context = gen.getOutputContext();
+
+        while (context != null) {
+            Object parent = context.getCurrentValue();
+            if (parent instanceof DruidQuery) {
```

Comment:
I don't see how this reduces readability any. Personally, I generally find it _more_ readable when things are broken out into helper methods, especially if there is duplication. 

First, the helper methods abstract away some of the "how" and replace it with "what." I find this good for two reasons: One, I generally find the "what" more interesting than the "how" when I'm starting to dig into some code. Two, I generally find it easier to understand the "how" when I already know the "what," then I do to derive the "what" from the "how."

Second, this makes it _immediately_ clear that this code is identical both here and in the other helper method. Without this method, I would have to read the two code blocks very closely in order to confirm that they are doing the same thing (as opposed to very subtly different things). That in and of itself is a _huge_ readability bonus.